sql/moz-fx-data-experiments/monitoring/experimenter_experiments_v1/query.py

```diff
@@ -145,7 +145,9 @@ def from_dict(cls, d) -> "ExperimentV6":
         converter = cattr.Converter()
         converter.register_structure_hook(
             datetime.datetime,
-            lambda num, _: datetime.datetime.fromisoformat(num.replace("Z", "+00:00")),
+            lambda num, _: pytz.utc.localize(
+                datetime.datetime.fromisoformat(num.replace("Z", "+00:00"))
+            ),
```

Comment:
this needs to use `normalize`, instead of `localize` to apply the timezone only if it was missing:
```suggestion
            lambda num, _: pytz.utc.normalize(
                datetime.datetime.fromisoformat(num.replace("Z", "+00:00"))
            ),
```

Reply:
I got `ValueError: Not naive datetime (tzinfo is already set)` before adding `localize()`. With `normalize()` I'm also getting `ValueError` 🤔

Reply:
ah, you're right, i messed up my local test, normalize does nothing of interest. I do, however, get `ValueError: Not naive datetime (tzinfo is already set)` for `num = datetime.datetime.utcnow().isoformat() + "Z"`

Reply:
this one works for me regardless of whether a `"Z"` suffix is present, but wouldn't handle a `"+00:00"` suffix:
```suggestion
            lambda num, _: datetime.datetime.fromisoformat(num.replace("Z", "") + "+00:00"),
```

Reply:
this one works for me with or without `"Z"` or `"+00:00"` suffix:
```suggestion
            lambda num, _: datetime.datetime.fromisoformat(num.replace("Z", "+00:00")).astimezone(pytz.utc),
```

Reply:
Yup this works. Thanks!